src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/BundlesRespondentService.java

```diff
@@ -152,4 +157,80 @@ public void addToBundlesCollection(CaseData caseData) {
             )
         );
     }
+
+    /**
+     * Removes a hearing bundle from the collection.
+     */
+    public void removeHearingBundles(CaseData caseData) {
+        String selectedParty = caseData.getRemoveBundleDropDownSelectedParty();
+
+        if ("selectClaimantHearingBundles".equals(selectedParty)) {
+            removeHearingBundles(caseData, caseData.getBundlesClaimantCollection());
+        } else if ("selectRespondentHearingBundles".equals(selectedParty)) {
+            removeHearingBundles(caseData, caseData.getBundlesRespondentCollection());
+        }
+    }
+
+    private void removeHearingBundles(CaseData caseData,
+                                      List<GenericTypeItem<HearingBundleType>> bundlesCollection) {
+
+        List<GenericTypeItem<RemovedHearingBundleItem>> removedHearingBundlesCollection =
+                caseData.getRemovedHearingBundlesCollection();
+        if (removedHearingBundlesCollection == null) {
+            removedHearingBundlesCollection = new ArrayList<>();
+            caseData.setRemovedHearingBundlesCollection(removedHearingBundlesCollection);
+        }
+
+        if (CollectionUtils.isEmpty(bundlesCollection)) {
+            return;
+        }
+
+        String selectedBundle = caseData.getRemoveHearingBundleSelect().getSelectedCode();
+
+        GenericTypeItem<HearingBundleType> bundleToRemove = bundlesCollection.stream()
+                .filter(bundle -> bundle.getId().equals(selectedBundle))
+                .findFirst()
+                .orElseThrow(() -> new NotFoundException("Bundle not found in the collection"));
+
+        bundlesCollection.removeIf(bundle -> bundle.getId().equals(selectedBundle));
+
+        DateTimeFormatter dateTimeFormatter = DateTimeFormat.forPattern("d MMMM yyyy HH:mm");
+        removedHearingBundlesCollection.add(
+            GenericTypeItem.from(RemovedHearingBundleItem.builder()
+                .bundleName(bundleToRemove.getValue().getUploadFile().getDocumentFilename())
+                .removedDateTime(dateTimeFormatter.print(DateTime.now()))
+                .removedReason(caseData.getHearingBundleRemoveReason())
+                .build()
+            )
+        );
+    }
+
+    public void populateSelectRemoveHearingBundle(CaseData caseData) {
+        String selectedParty = caseData.getRemoveBundleDropDownSelectedParty();
+        if ("selectClaimantHearingBundles".equals(selectedParty)) {
```

Comment:
May be using a constant for "selectClaimantHearingBundles" as it is used here and on lines 167/169. Just for consistency.

Reply:
resolved https://github.com/hmcts/et-ccd-callbacks/pull/1878/commits/2205aded8dfae4f6851e9a2336d8c9b30056a046